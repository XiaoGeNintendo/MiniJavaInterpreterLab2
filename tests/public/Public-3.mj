

// explicit constructor

class A {
    int a;
    string b;

    A() {
        // default value for a and b
    }

    A(int a, string b) {
        this.a = a;
        this.b = b;
    }
}
int main() {
    A a1 = new A();

    println(a1.a);
    println(a1.b);

    A a2 = new A(20, "world");
    println(a2.a);
    println(a2.b);

    return 0;
}