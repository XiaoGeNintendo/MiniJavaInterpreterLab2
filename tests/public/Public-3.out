Process exits with the code 33.
